packages/completer/src/default/connector.ts

```diff
@@ -73,31 +73,22 @@ namespace Private {
    * duplicates from the context list that appear in the kernel list.
    */
   export function mergeReplies(
-    kernel: CompletionHandler.IReply,
-    context: CompletionHandler.IReply
-  ): CompletionHandler.IReply {
-    // If one is empty, return the other.
-    if (kernel.matches.length === 0) {
+    kernel: CompletionHandler.ICompletionItemsReply,
+    context: CompletionHandler.ICompletionItemsReply
+  ): CompletionHandler.ICompletionItemsReply {
```

Comment:
I feel that we should remove this special handling/merging of replies and just have separate completion providers for `context` and `kernel`.